### ISSUE TITLE
[NHentai] Tag exclusion with "-" instead of space doesn't work as intended

### ISSUE THREAD

**Issue by hyoretsu** [TheNetsky/community-extensions#13]
My last issue, #11, after some digging into it, was because I'm excluding the `gender-bender` tag. Messing with the API, this query `https://nhentai.net/api/galleries/search?query=-gender-bender` returns nothing. Including that tag, like `https://nhentai.net/api/galleries/search?query=+gender-bender` works.

Searching for `https://nhentai.net/api/galleries/search?query=alo` returns something, but searching for `https://nhentai.net/api/galleries/search?query=alo%20-gender-bender` returns nothing. Then, if we search for `https://nhentai.net/api/galleries/search?query=alo%20-"gender bender"` returns what we want, since the original query didn't have anything with that tag to begin with.

In short, what I think is that there's a problem with asking the users to switch a space for a dash. It seems to work when including those tags, but not when excluding them. In Tachiyomi, we had to do `-"gender bender"` like normal. In iOS' case, we'd have to treat their weird quote ” first.

**Comment by henderea** [comment]
It should be possible to replace the fancy quotes with regular quotes when saving the settings or running the search. Though it would be best if the app disabled fancy quotes. 

I think turning off the "smart punctuation" toggle in keyboard settings in the system settings app may help on an individual device though.

**Comment by hyoretsu** [comment]
Same issue (sorta) happens with normal dash tags, like `non-h`

`https://nhentai.net/api/galleries/search?query=alo`
Returns something
`https://nhentai.net/api/galleries/search?query=alo%20-non-h`
Returns other things
`https://nhentai.net/api/galleries/search?query=alo%20-"non-h"`
Returns the same something as the first one

**Comment by hyoretsu** [comment]
In short, it's  just needed to automatically fix iOS fancy quote and change the instructions to use `-"tag"` for tags with a dash or space. (If my observations are correct)

Tested locally and it's working fine with no other changes.

**Comment by TheNetsky** [comment]
Feel free to make a PR.
